pkg/k8shandler/certificates.go

```diff
@@ -2,138 +2,81 @@ package k8shandler
 
 import (
 	"fmt"
+	"io/ioutil"
 	"os/exec"
+	"path"
+	"sync"
 
 	"github.com/ViaQ/logerr/log"
 	"github.com/openshift/cluster-logging-operator/pkg/constants"
 	"github.com/openshift/cluster-logging-operator/pkg/utils"
 	"k8s.io/apimachinery/pkg/api/errors"
-
-	"k8s.io/apimachinery/pkg/util/sets"
 )
 
-var deprecatedKeys = sets.NewString("app-ca", "app-key", "app-cert", "infra-ca", "infra-key", "infra-cert")
-
-// golang doesn't allow for const maps
-var secretCertificates = map[string]map[string]string{
-	"master-certs": {
-		"masterca":  "ca.crt",
-		"masterkey": "ca.key",
-	},
-	"elasticsearch": {
-		"elasticsearch.key": "elasticsearch.key",
-		"elasticsearch.crt": "elasticsearch.crt",
-		"logging-es.key":    "logging-es.key",
-		"logging-es.crt":    "logging-es.crt",
-		"admin-key":         "system.admin.key",
-		"admin-cert":        "system.admin.crt",
-		"admin-ca":          "ca.crt",
-	},
-	"kibana": {
-		"ca":   "ca.crt",
-		"key":  "system.logging.kibana.key",
-		"cert": "system.logging.kibana.crt",
-	},
-	"kibana-proxy": {
-		"server-key":     "kibana-internal.key",
-		"server-cert":    "kibana-internal.crt",
-		"session-secret": "kibana-session-secret",
-	},
-	"curator": {
-		"ca":       "ca.crt",
-		"key":      "system.logging.curator.key",
-		"cert":     "system.logging.curator.crt",
-		"ops-ca":   "ca.crt",
-		"ops-key":  "system.logging.curator.key",
-		"ops-cert": "system.logging.curator.crt",
-	},
-	"fluentd": {
-		"ca-bundle.crt": "ca.crt",
-		"tls.key":       "system.logging.fluentd.key",
-		"tls.crt":       "system.logging.fluentd.crt",
-	},
-}
+var mutex sync.Mutex
 
-func (clusterRequest *ClusterLoggingRequest) extractSecretToFile(secretName string, key string, toFile string) (err error) {
-	secret, err := clusterRequest.GetSecret(secretName)
+func (clusterRequest *ClusterLoggingRequest) extractMasterCerts() (err error) {
+	secret, err := clusterRequest.GetSecret(constants.MasterCASecretName)
```

Comment:
When the new code runs for the first time after an upgrade, it'll dump files "masterca" and "masterkey" on disk and keep them circulating. Just might account for that and skip them altogether.